### PR TITLE
[HPT-1086] Adds search service to iiif manifest.

### PR DESCRIPTION
**Pull request by andjsmit**
Also changes canvas IDs to use available parameters contained by both search and manifest code.

### CODE REVIEW

File: app/services/manifest_builder/services_builder.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      @record = record
+    end
+
+    def apply(manifest)
+      return if
+        record.class == AllCollectionsPresenter ||
+        record.class == CollectionShowPresenter
+      service_array = {
+        "@context"  => "http://iiif.io/api/search/0/context.json",
+        "@id"       => "http://localhost:3000/search/#{record.id}",
```
Comment (mwoodiupui) [inline app/services/manifest_builder/services_builder.rb]: I'm concerned about the hard-coded host and port.

Comment (andjsmit) [reply app/services/manifest_builder/services_builder.rb]: Ooops. Thought I got that out of there. Was using it for testing. Will correct. Thx.

**Review by wang58 — COMMENTED** [review]
I saw the search bar showed up and most of the word got the correct highlight. But for some words,  it seems found the word , but highlights lost. We might take another look? (I used my own newspaper)

**Comment by andjsmit** [comment]
As discussed on slack, the issue is with searching two-letter words. SOLR has them but the JSON word boundary file does not. I would like to leave this alone for now and write-up another story specifically dealing with two-letter words. With this PR, the behavior when searching 2-letters words should be a recognition that they exist on a page but are not highlighted.

**Review by aploshay — APPROVED** [review]
+1

**Comment by wang58** [comment]
Agree to have another story for two-letter word search. Otherwise, the highlight works great. Good to merge.  +1

**Pull request merged**